packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -461,3 +487,78 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
+  'moveForward': const ExtendSelectionByCharacterIntent(forward: true, collapseSelection: true),
+  'moveBackward': const ExtendSelectionByCharacterIntent(forward: false, collapseSelection: true),
+
+  'moveUp': const ExtendSelectionVerticallyToAdjacentLineIntent(forward: false, collapseSelection: true),
+  'moveDown': const ExtendSelectionVerticallyToAdjacentLineIntent(forward: true, collapseSelection: true),
+
+  'moveLeftAndModifySelection': const ExtendSelectionByCharacterIntent(forward: false, collapseSelection: false),
+  'moveRightAndModifySelection': const ExtendSelectionByCharacterIntent(forward: true, collapseSelection: false),
+  'moveUpAndModifySelection': const ExtendSelectionVerticallyToAdjacentLineIntent(forward: false, collapseSelection: false),
+  'moveDownAndModifySelection': const ExtendSelectionVerticallyToAdjacentLineIntent(forward: true, collapseSelection: false),
+
+  'moveWordLeft': const ExtendSelectionToNextWordBoundaryIntent(forward: false, collapseSelection: true),
+  'moveWordRight': const ExtendSelectionToNextWordBoundaryIntent(forward: true, collapseSelection: true),
+  'moveToBeginningOfParagraph': const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
+  'moveToEndOfParagraph': const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
+
+  'moveWordLeftAndModifySelection': const ExtendSelectionToNextWordBoundaryIntent(forward: false, collapseSelection: false),
+  'moveWordRightAndModifySelection': const ExtendSelectionToNextWordBoundaryIntent(forward: true, collapseSelection: false),
+  'moveParagraphBackwardAndModifySelection': const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false),
+  'moveParagraphForwardAndModifySelection': const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: false),
+
+  'moveToLeftEndOfLine': const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
+  'moveToEndEndOfLine': const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
+  'moveToBeginningOfDocument': const ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: true),
+  'moveToEndOfDocument': const ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: true),
+
+  'moveToLeftEndOfLineAndModifySelection': const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false),
+  'moveToRightEndOfLineAndModifySelection': const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: false),
+  'moveToBeginningOfDocumentAndModifySelection': const ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: false),
+  'moveToEndOfDocumentAndModifySelection': const ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: false),
+
+  'transpose': const TransposeCharactersIntent(),
+
+  // TODO(knopp): Page Up/Down intents are missing
```

Comment:
I opened a PR for pageup/down in text editing here: https://github.com/flutter/flutter/pull/105497